src/lib/merkle_mask/masking_merkle_tree.ml

```diff
@@ -333,6 +359,9 @@ module Make (Inputs : Inputs_intf.S) = struct
       let old_root_hash = merkle_root t in
       let account_data = Location_binable.Table.to_alist t.account_tbl in
       Base.set_batch (get_parent t) account_data ;
+      Option.iter t.next_available_token ~f:(fun tid ->
+          Base.set_next_available_token (get_parent t) tid ;
```

Comment:
what consequence does setting the parent's next available have for other masks with the same parent?

Reply:
Ah. They should also be updated. I'll re-jig this to take the maximum of the mask's `next_available_token` and the parent's.